include/simics/dmllib.h

```diff
@@ -631,6 +631,27 @@ _simple_event_predicate(lang_void *_data, lang_void *match_data) {
     return 0;
 }
 
+/*
+ * The modern serialized representation of after event data is a
+ * pseudo-dictionary, containing entries for method indices, arguments, and
+ * domains (associated objects for cancellation):
+ * [["indices", [index0, index1, ...]],
+ *  ["arguments", [arg0, arg1, ...]],
+ *  ["domains", [domain0, domain1, ...]]
+ * ]
+ * The entries may appear in any order, and any entry may be omitted if not
+ * applicable, (for example, a method call has no indices/arguments or
+ * associated objects). If all entries may be omitted, then Nil is used
```

Comment:
`s/may be/are/` ?

I think lowercase `nil` is the convention.

Reply:
I disagree with `s/may be/are/`; "may be" ties back to the previous sentence. "A dict is like this, but an entry _may be_ omitted if not applicable, and if every entry _may be_ omitted, then the dict is nil instead.`

You're right, lowercase `nil` it is.

Reply:
Then "if no entry is applicable" would sound more direct?

Still, I find it a strange mix of conditional and certain stuff: If one single entry is unapplicable, then it *may be* omitted, while if all entries are unapplicable, then the dict *is* replaced with nil. I'd either say "is" on both or "may be" on both (I suppose the truth is "is" on serialization and "may be" on deserialization).